src/shutdown/detach-dm.c

```diff
@@ -99,23 +99,31 @@ static int delete_dm(DeviceMapper *m) {
         assert(major(m->devnum) != 0);
         assert(m->path);
 
-        fd = open("/dev/mapper/control", O_RDWR|O_CLOEXEC);
-        if (fd < 0)
-                return -errno;
-
-        r = fsync_path_at(AT_FDCWD, m->path);
-        if (r < 0)
-                log_debug_errno(r, "Failed to sync DM block device %s, ignoring: %m", m->path);
-
-        return RET_NERRNO(ioctl(fd, DM_DEV_REMOVE, &(struct dm_ioctl) {
+        struct dm_ioctl dminfo = {
                 .version = {
                         DM_VERSION_MAJOR,
                         DM_VERSION_MINOR,
                         DM_VERSION_PATCHLEVEL
                 },
                 .data_size = sizeof(struct dm_ioctl),
                 .dev = m->devnum,
-        }));
+        };
```

Comment:
why move this up? please place this right before the first time you are going to use this.